### ISSUE TITLE
Getting a CastError for double values

### ISSUE THREAD

**Issue by MagnusJohansson** [dart-ogurets/dart-openapi-maven#70]
In my OpenAPI json I have a couple of properties that are defined as numbers/doubles:

``` json
          "conditionAmount": {
                        "type": "number",
                        "description": "The condition amount.",
                        "format": "double"
                    },
```

With the generated contract class, I get this cast error (this value happen to be a whole number; 10)
``` dart
    {
      final _jsonData = json[r'conditionAmount'];
      if (_jsonData == null)
        throw DeserialisationError(json, r'conditionAmount', r'',
            'conditionAmount field is null and is required to have a value');
      conditionAmount = (_jsonData.to as double);
    }
```
![image](https://user-images.githubusercontent.com/674909/139136703-0dcceb70-c35a-472d-ac2e-c3f9b4baaa9f.png)

If I change these "as double" casts to __jsonData.toDouble(), it seems to work just fine.
``` dart
   {
      final _jsonData = json[r'conditionAmount'];
      if (_jsonData == null)
        throw DeserialisationError(json, r'conditionAmount', r'',
            'conditionAmount field is null and is required to have a value');
      conditionAmount = (_jsonData.toDouble());  <---  
    }

```

**Comment by rvowles** [comment]
Yup - I can see how that would happen as Dart is just going to convert 12 into an int. Really the original serializer should be sending 12.0 so its not honouring the contract but we should be more tolerant. Its going to take a bit of work because that code just does a generic _jsonData as {{classname}}

**Comment by rvowles** [comment]
I have added this schema:

`    DoubleAndIntConversion:
      type: object
      properties:
        intList:
          type: array
          items:
            type: integer
        intMap:
          type: object
          additionalProperties:
            type: integer
        doubleList:
          type: array
          items:
            type: number
            format: double
        doubleMap:
          type: object
          additionalProperties:
            type: number
            format: double`

and this test:

`  test('double with ints in them tests and vs versa', () {
    const testData = {
      'intList': [1, 2.6],
      'intMap': {'one': 1, 'two': 2.7},
      'doubleList': [1, 2.6],
      'doubleMap': {'one': 1, 'two': 2.7},
    };

    final data = DoubleAndIntConversion.fromJson(testData);
    expect(data.intList, [1, 2]);
    expect(data.intMap, {'one': 1, 'two': 2});
    expect(data.doubleList, [1.0, 2.6]);
    expect(data.doubleMap, {'one': 1.0, 'two': 2.7});
  });`

into the integration tests for both variants (null-safe and non-null-safe)

**Comment by rvowles** [comment]
This will be resolved in the 6.1 release
